plugins/modules/cloud/misc/proxmox_disk.py

```diff
@@ -0,0 +1,704 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2022, Castor Sky (@castorsky) <csky57@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: proxmox_disk
+short_description: Management of a disk of a Qemu(KVM) VM in a Proxmox VE cluster.
+version_added: 5.4.0
+description:
+  - Allows you to perform some supported operations on a disk in Qemu(KVM) Virtual Machines in a Proxmox VE cluster.
+author: "Castor Sky (@castorsky) <csky57@gmail.com>"
+options:
+  name:
+    description:
+      - The (unique) name of the VM.
+      - Required only one of C(name) or C(vmid).
+    type: str
+  vmid:
+    description:
+      - The (unique) ID of the VM.
+    type: int
+  disk:
+    description:
+      - The disk key (C(unused[n]), C(ide[n]), C(sata[n]), C(scsi[n]) or C(virtio[n])) you want to operate on.
+    type: str
+    required: True
+  state:
+    description:
+      - Indicate desired state of the disk. C(absent) can be used only with C(remove) action
+    type: str
+    choices: ['present', 'absent']
+    default: present
+  action:
+    description:
+      - Action to perform on the disk.
+    type: str
+    choices: ['create', 'update', 'grow', 'detach', 'move', 'remove']
```

Comment:
I am a bit confused by this. What if I call the module with:
```yaml
  - c.g.proxmox_disk:
      state: present
      action: remove
```
will it add a disk or remove a disk?

From here it looks like we should remove the `state` parameter as it is and rename `action` to be `state` and wordsmith the choices' names a bit to reflect that change.

Reply:
Generally modules should work with states, not with actions. That's what most Ansible modules do, and what users expect from modules.

Reply:
> will it add a disk or remove a disk?
It will fail with msg "Unsupported combination of parameters state and action"
But anyway
> Generally modules should work with states, not with actions
I will edit module to exclude action parameter

Reply:
> will it add a disk or remove a disk?

It will fail with msg "Unsupported combination of parameters state and action"
But anyway

> Generally modules should work with states, not with actions

I will edit module to exclude action parameter

Reply:
Action parameter dropped